pre_processing/read_era5_jasmin.F90

```diff
@@ -8,246 +8,339 @@
 !
 !-------------------------------------------------------------------------------
 
-subroutine read_era5_jasmin(nwp_path, ecmwf, preproc_dims, preproc_geoloc, &
-     preproc_prtm, verbose, nwp_flag)
+subroutine read_era5_jasmin_wind_nc(ecmwf, nwp_fnames, idx, nwp_flag)
 
-   use orac_ncdf_m
    use preproc_constants_m
-   use preproc_structures_m
+   use preproc_structures_m, only: preproc_nwp_fnames_t
 
    implicit none
 
-   character(len=*),        intent(in)    :: nwp_path
-   type(ecmwf_t),           intent(in)    :: ecmwf
-   type(preproc_dims_t),    intent(in)    :: preproc_dims
-   type(preproc_geoloc_t),  intent(in)    :: preproc_geoloc
-   type(preproc_prtm_t),    intent(inout) :: preproc_prtm
-   logical,                 intent(in)    :: verbose
-   integer,                 intent(in)    :: nwp_flag
-
-   integer(lint),     external            :: INTIN, INTOUT, INTF
-   integer(lint),     parameter           :: BUFFER = 2000000
-
-   integer(lint),            dimension(1) :: intv, old_grib, new_grib
-   real(dreal)                            :: grid(2), area(4)
-   real(dreal), allocatable, dimension(:) :: old_data, new_data
-   character(len=20),        dimension(1) :: charv
-
-   real(sreal),       pointer             :: array2d(:,:), array3d(:,:,:)
-   integer(4)                             :: n, ni, nj, i, j, k, ivar
-   integer(4)                             :: fid, nvar
-   integer(4)                             :: old_len, new_len
-   character(len=20)                      :: name
-   logical                                :: three_d
-   real(sreal)   :: dummy2d(ecmwf%xdim,ecmwf%ydim,1,1)
-   real(sreal)   :: dummy3d(ecmwf%xdim,ecmwf%ydim,ecmwf%kdim,1)
-   real(sreal)   :: dummy3d_2(ecmwf%xdim,ecmwf%ydim,ecmwf%kdim)
-#ifdef WRAPPER
-   real(sreal)   :: ecmwf_lon(ecmwf%xdim)
-   real(sreal)   :: ecmwf_lat(ecmwf%ydim)
-   integer(lint) :: pointer_x(preproc_dims%min_lon:preproc_dims%max_lon)
-   integer(lint) :: pointer_y(preproc_dims%min_lat:preproc_dims%max_lat)
-   real(sreal)   :: diff_lon(ecmwf%xdim), diff_lat(ecmwf%ydim)
+   type(ecmwf_t),    intent(inout)        :: ecmwf
+   type(preproc_nwp_fnames_t), intent(in) :: nwp_fnames
+   integer,          intent(in)           :: idx
+   integer,          intent(in)           :: nwp_flag
+
+   call ecmwf_wind_init(ecmwf)
+   call ecmwf_abvec_init(ecmwf)
+
+   ! loop over given files
```

Comment:
There ain't a loop. Just nice, tidy reads

Reply:
Err, yes. Can you tell I copy-pasted that code from elsewhere?
Fixed.